public/pages/DetectorResults/utils/utils.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+import { Detector } from '../../../models/interfaces';
+import {
+  NO_FULL_SHINGLE_ERROR_MESSAGE,
+  NO_DATA_IN_WINDOW_ERROR_MESSAGE,
+  NO_RCF_MODEL_ERROR_MESSAGE,
+} from './constants';
+import { DETECTOR_STATE, SHINGLE_SIZE } from '../../../utils/constants';
+import { DETECTOR_INIT_FAILURES } from '../../../pages/DetectorDetail/utils/constants';
+import moment, { Moment } from 'moment';
+
+export const IS_INIT_OVERTIME_FIELD = 'isInitOvertime';
+export const INIT_DETAILS_FIELD = 'initDetails';
+export const INIT_ERROR_MESSAGE_FIELD = 'initErrorMessage';
+export const INIT_ACTION_ITEM_FIELD = 'initActionItem';
+
+export const getDetectorInitializationInfo = (detector: Detector) => {
+  const currentTime = moment();
+
+  let result = {
+    [IS_INIT_OVERTIME_FIELD]: false,
+    [INIT_DETAILS_FIELD]: {},
+  };
+  if (isDetectorInitOverTime(currentTime, detector)) {
+    result[IS_INIT_OVERTIME_FIELD] = true;
+    result[INIT_DETAILS_FIELD] = getInitOverTimeDetails(detector);
+  }
+  return result;
+};
+
+const isDetectorInitOverTime = (currentTime: Moment, detector: Detector) => {
+  return (
+    detector &&
+    detector.curState === DETECTOR_STATE.INIT &&
+    detector.initializationError &&
+    !detector.initializationError.includes(NO_RCF_MODEL_ERROR_MESSAGE) &&
+    //@ts-ignore
+    currentTime
+      .subtract(
+        SHINGLE_SIZE * detector.detectionInterval.period.interval,
+        detector.detectionInterval.period.unit.toLowerCase()
+      )
+      //@ts-ignore
+      .valueOf() > detector.enabledTime
+  );
+};
+
+const getInitOverTimeDetails = (detector: Detector) => {
+  let result = {
+    [INIT_ERROR_MESSAGE_FIELD]: '',
+    [INIT_ACTION_ITEM_FIELD]: '',
+  };
+  if (!detector.initializationError) {
+    return result;
+  }
+  if (detector.initializationError.includes(NO_FULL_SHINGLE_ERROR_MESSAGE)) {
+    result[INIT_ERROR_MESSAGE_FIELD] = 'collected data is insufficient';
+    result[INIT_ACTION_ITEM_FIELD] =
+      DETECTOR_INIT_FAILURES.NO_TRAINING_DATA.actionItem;
```

Comment:
small: should there be a separate `NO_FULL_SHINGLE` enum instead of `NO_TRAINING_DATA` here? It feels like `NO_TRAINING_DATA` should be reserved for the `NO_DATA_IN_WINDOW_ERROR_MESSAGE` below this

Reply:
I think for now we don't treat long initialization as a initialization failure case, instead we think it is somewhere between initialization and initialization failure, same as `No_FULL_SHINGLE` . That is why I did not add `No_FULL_SHINGLE`  to `DETECTOR_INIT_FAILURES`. 

And regarding the `actionItem`, I use NO_TRAINING_DATA's actionItem as I think it can work for `No_FULL_SHINGLE` case, because both `NO_TRAINING_DATA` and `No_FULL_SHINGLE` has a common point: no enough data is collected, while `NO_DATA_IN_WINDOW_ERROR_MESSAGE`'s point is no data at all. Besides, `NO_TRAINING_DATA`'s actionItem is to let user ingest enough data or adjust detectionInterval, this can help solve `No_FULL_SHINGLE` issue as well.

Reply:
I see, thanks for the explanation. That makes sense.